packages/opentelemetry-node/src/instrumentation/PluginLoader.ts

```diff
@@ -83,6 +83,18 @@ export class PluginLoader {
         return this;
       }
 
+      const alreadyRequiredModules = Object.keys(require.cache);
+      const requiredModulesToHook = modulesToHook.filter(name =>
+        alreadyRequiredModules.some(cached => cached.includes(name))
+      );
+      if (requiredModulesToHook.length > 0) {
+        this.logger.warn(
+          `Some modules (${requiredModulesToHook.join(
+            ', '
+          )}) were already required when their respective plugin was loaded, some plugins might not work. Make sure the SDK is setup before you require in other modules.`
+        );
+      }
+
       // Enable the require hook.
       hook(modulesToHook, (exports, name, baseDir) => {
```

Comment:
if some of the modules have been already required, shouldn't those modules be then removed from `pluginsToLoad` ?

if that is true I'm wondering if this filtering / finding could be refactored and became a one loop only, where you can filter based on `enabled`, `path` and then if modules has been already required - just set a boolean flag to raise the warning later.

Reply:
Its complicated, it depend on how the user required the module before us. If he didn't use the module it should work fine, ex:

```
import express from 'express'
import ot from '@opentelemetry/node`
ot.getTracer()

const app = express()
```

However if he create the express app **before** requiring the tracer, it **might** (somes patch still works in those case because it depend on the 3rd party module implementatoin) not work at all.

Sadly performing this analysis is hard (ie: not worth IMO) since it will need to be defined for each plugin and possibility for each version of the module each plugin patch.